### PR TITLE
TOOLS-2588: [v4.2] sslAllowInvalidHostnames bypass ssl/tls server cer…

### PR DESCRIPTION
**Pull request by huan-Mongo**
…tification validation entirely

Tool's v4.2-master points to MTC v2.0.3 according to here https://github.com/mongodb/mongo-tools/blob/v4.2-master/Gopkg.lock#L151
I created MTC branch v4.2 branch based on v2.0.3 tag. This branch will serve Tools v4.2-master backport purposes only. 

I don't have the privilege to add Evergreen monitoring on v4.2 branch for now. So I manually generate the patch build https://evergreen.mongodb.com/version/5f4012815623434841921e3b

I made change to the testing CA files as well since there were errors in the first patch build 
```
 [2020/08/21 18:18:23.826] 2020-08-21T18:18:23.826+0000 F NETWORK  [main] The provided SSL certificate is expired or not yet valid.
 [2020/08/21 18:18:23.826] 2020-08-21T18:18:23.826+0000 F -        [main] Fatal Assertion 28652 at src/mongo/util/net/ssl_manager_openssl.cpp 973
 [2020/08/21 18:18:23.826] 2020-08-21T18:18:23.826+0000 F -        [main]
 [2020/08/21 18:18:23.826] ***aborting after fassert() failure
```
https://evergreen.mongodb.com/task/mongo_tools_common_ubuntu1604_ssl_integration_4.0_patch_0b00b4fe7c54276f166f795c7d66b30a523f08dc_5f400eb1562343484191c4fc_20_08_21_18_13_23

### CODE REVIEW

**Review by tfogo — APPROVED** [review]
LGTM!

**Comment by tfogo** [comment]
By the way, should we tag this change as v2.1.0 so tools 4.2 can point to a specific version? I believe a minor version bump from 2.0.3 to 2.1.0 would be appropriate here.

**Comment by huan-Mongo** [comment]
@tfogo Cool! That's a good idea! I will tag it as 2.1.0 after merged.

File: .evergreen.yml [hunk 1]
```diff
@@ -15,15 +15,15 @@ mongo_tools_variables:
       mongod_args: ""
       mongod_port: 33333
     ssl: &mongod_ssl_startup_args
-      mongod_args: "--sslMode requireSSL --sslCAFile db/testdata/ca.pem --sslPEMKeyFile db/testdata/server.pem"
+      mongod_args: "--sslMode requireSSL --sslCAFile ./db/testdata/ca-ia.pem --sslPEMKeyFile ./db/testdata/test-server.pem"
```
Comment (pmeredit) [inline .evergreen.yml:18]: was it not working without the `./`?

Comment (huan-Mongo) [reply .evergreen.yml:18]: it seems working before. But I changed it to a new certificate here.

**Pull request merged**